src/add-route-callbacks.js

```diff
@@ -1,19 +1,168 @@
 const createCallback = require('./create-callback')
 const loadFile = require('./load-file')
 const path = require('path')
+const fileExists = require('./file-exists')
 
 function addRouteCallbacks (routeObjects, routesDir) {
-  return routeObjects.map(routeObj => {
+  const objects = []
+  routeObjects.forEach(routeObj => {
+    let Route
+
+    // implicit routes
+
     if (routeObj.children.length > 0) {
-      routeObj.children = addRouteCallbacks(routeObj.children, path.join(routesDir, routeObj.name))
+      if (fileExists(`${routeObj.name}/index.js`, routesDir)) {
```

Comment:
Easy enough to just iterate over the whitelisted verbs instead of copy/paste.

```js
const allowedMethods = ['get', 'put', 'post', 'patch', 'delete', 'options', 'head']
allowedMethods.forEach(method => {
  let Route
  if (method.toLower() === 'get') {
   Route = loadFile(`${this.routeObj.name}/index.js`)
  }
  if (!Route) {
    Route = loadFile(`${this.routeObj.name}/index.${method}.js`)
  }
  if (Route) {
    objects.push({
      method: method,
      callback: createCallback(Route),
      name: 'index',
      path: routeObj.path
    })
  }
})
```

Either that or use something like https://www.npmjs.com/package/glob to read all of the applicable files at once and then parse into the objects.